Mathlib/SetTheory/Surreal/Division.lean

```diff
@@ -0,0 +1,714 @@
+      simp only
+      have := eq2 h3 h4 h5 inv_l inv_r inv_numeric x_pos (InvTy.right₂ i j)
+      simp only [components] at this
+      conv_lhs at this => {
+        pattern (occs := *) (PGame.mk l r L R).inv'.moveRight j
+        simp [inv']
+      }
+      rw [this]
+      simp only [add_lt_iff_neg_left]
+      apply mul_neg_of_pos_of_neg
+      · exact right_pos_of_pos h5 x_pos i
+      · apply sub_neg.mpr
+        have := mk_lt_mk_moveRight inv_numeric (InvTy.right₂ i j)
+        conv_rhs at this => simp [inv']
+        exact this
+
+lemma onag_1_10_iii_right' {l r} {L : l → PGame} {R : r → PGame}
+    (h3 : ∀ i, 0 < (L i) → (L i).inv'.Numeric)
+    (h4 : ∀ j, (R j).inv'.Numeric) (h5 : (PGame.mk l r L R).Numeric)
+    (inv_l : ∀ (i : { i // 0 < L i }), mk (L i) (h5.2.1 i) * mk (L i).inv' (h3 i i.2) = 1)
+    (inv_r : ∀ j, mk (R j) (h5.2.2 j) * mk (R j).inv' (h4 j) = 1)
+    (inv_numeric : (PGame.mk l r L R).inv'.Numeric)
+    (x_pos: 0 < PGame.mk l r L R)
+    (ij : Sum ({ i // 0 < L i } × InvTy { i // 0 < L i } r true)
+      (r × InvTy { i // 0 < L i } r false)) :
+    let x := mk (PGame.mk l r L R) h5
+    let y := mk ((PGame.mk l r L R).inv') inv_numeric
+    match ij with
+    | Sum.inl ⟨i, j⟩ =>
+      let x' := mk (L i) (h5.2.1 i)
+      let y' := mk (invVal (fun (i : {i // 0 < L i}) => L i) R (fun i => (L i).inv')
+        (fun j => (R j).inv') (PGame.mk l r L R) j)
+        (by apply invVal_numeric <;> tauto)
+      1 < x' * y + x * y' - x' * y'
+    | Sum.inr ⟨i, j⟩ =>
+      let x' := mk (R i) (h5.2.2 i)
+      let y' := mk (invVal (fun (i : {i // 0 < L i}) => L i) R (fun i => (L i).inv')
+        (fun j => (R j).inv') (PGame.mk l r L R) j)
+        (by apply invVal_numeric <;> tauto)
+      1 < x' * y + x * y' - x' * y' := by
+  rcases ij <;> simp only
+  · case _ val =>
+      rcases val with ⟨i, j⟩
+      simp only
+      have := eq2 h3 h4 h5 inv_l inv_r inv_numeric x_pos (InvTy.left₂ i j)
+      simp only [components] at this
+      conv_lhs at this => {
+        pattern (occs := *) (PGame.mk l r L R).inv'.moveRight j
+        simp [inv']
+      }
+      rw [this]
+      simp only [lt_add_iff_pos_right]
+      apply Left.mul_pos
+      · exact i.2
+      · apply sub_pos.mpr
+        have := mk_moveLeft_lt_mk inv_numeric (InvTy.left₂ i j)
+        conv_rhs at this => simp [inv']
+        exact this
+  · case _ val =>
+      rcases val with ⟨i, j⟩
+      simp only
+      have := eq2 h3 h4 h5 inv_l inv_r inv_numeric x_pos (InvTy.left₁ i j)
+      simp only [components] at this
+      conv_lhs at this => {
+        pattern (occs := *) (PGame.mk l r L R).inv'.moveLeft j
+        simp [inv']
+      }
+      rw [this]
+      simp only [lt_add_iff_pos_right]
+      apply Left.mul_pos
+      · exact right_pos_of_pos h5 x_pos i
+      · apply sub_pos.mpr
+        have := mk_moveLeft_lt_mk inv_numeric (InvTy.left₁ i j)
+        conv_rhs at this => simp [inv']
+        exact this
+
+/-! ### The main lemma -/
+
+lemma onag_1_10 :
```

Comment:
`numeric_inv'_and_inv_surreal`? Honestly it's probably fine to just name this one `main`.